src/sentry/services/hybrid_cloud/user/service.py

```diff
@@ -195,12 +192,11 @@ def get_user_avatar(self, *, user_id: int) -> RpcAvatar | None:
 
 
 @back_with_silo_cache("user_service.get_user", SiloMode.REGION, RpcUser)
-def get_user(user_id: int) -> RpcUser:
+def get_user(user_id: int) -> RpcUser | None:
     users = user_service.get_many(filter=dict(user_ids=[user_id]))
     if len(users) > 0:
         return users[0]
-    else:
-        return RpcUser(is_anonymous=True)
+    return None
```

Comment:
Supporting optional values in cached service calls is a significant change, but will allow us to use caching in more places. Resources like subscriptions and integrations don't have simple nullobjects that we can use, and supporting `None` lets caching work well with the optional returns we have in many RPC services.

Reply:
👍

Reply:
I am generally ok with this -- that said, is there any chance we might consider a `NullWrapper` or sentinel?

Reason is to help distinguish "the data isn't there yet" from "there data is definitely None".  But it's a minor consideration.

Reply:
Getting a NullObject that works with the typings could be fun. I'll see if I can get that working in a follow up change.